### ISSUE TITLE
Incorrect TypeScript definitions

### ISSUE THREAD

**Issue by davidxmoody** [react-community/lottie-react-native#398]
### Description

Importing with `import LottieView from "lottie-react-native"` gives types that compile but `LottieView` is actually undefined.

### Steps to Reproduce

```typescript
import * as React from "react"
import LottieView from "lottie-react-native"

const lottieAnimation = require("./_lottie-loading-spinner.json")

export default class LoadingSpinner extends React.PureComponent {
  public render() {
    return <LottieView source={lottieAnimation} autoPlay loop />
  }
}
```

**Expected behavior:** The above code should run fine (it compiles).

**Actual behavior:** Crash with `Invariant Violation: Element type is invalid: expected a string`

<img width="377" alt="screen shot 2018-10-25 at 10 45 19" src="https://user-images.githubusercontent.com/4699197/47491847-95a7a300-d843-11e8-9043-72bde1621398.png">

### Versions

`"lottie-react-native": "^2.5.9"`

The following works correctly:

```typescript
import * as React from "react"
const LottieView = require("lottie-react-native")

const lottieAnimation = require("./_lottie-loading-spinner.json")

export default class LoadingSpinner extends React.PureComponent {
  public render() {
    return <LottieView source={lottieAnimation} autoPlay loop />
  }
}
```

This appears to be the same thing as described in https://github.com/react-community/lottie-react-native/issues/271 and https://github.com/react-community/lottie-react-native/pull/291 which has come back.

**Comment by ljharb** [comment]
This isn’t due to TypeScript or type devs (which don’t cause runtime behavior) but that LottieView is undefined here.

You’re probably using TypeScript with its default settings, which have a broken module system. Try enabling esModuleInterop and syntheticImports.

**Comment by leighman** [comment]
https://github.com/react-community/lottie-react-native/commit/39e45d1733de1bc2ab5d79e6422b9aa4b56d2756 is the commit that changed this quite recently.

The `index.js` doesn't have a default export so the `index.d.ts` is a lie. Typescript with its default settings has a mechanism to support commonjs modules which is `export = Blah` and `import Blah = require("blah")`

**Comment by ljharb** [comment]
Indeed, #389 looks incorrect to me. But the error reported in this OP has nothing to do with type definitions.

**Comment by rodrigoelp** [comment]
Hello @davidxmoody!

Could you please share your project configuration? I've created a [demo project](https://github.com/rodrigoelp/lottie-rn-typescript-demo). If you open the file `src/index.tsx` you should be able to see I am using the same code.

**Comment by rodrigoelp** [comment]
By the way @davidxmoody, you should be able to check out the tsconfig file as well, plain defaults.

**Comment by leighman** [comment]
@rodrigoelp that does work because of `esModuleInterop` but the index.d.ts is still wrong. It seems to also work fine with `export = LottieView`

**Comment by rodrigoelp** [comment]
Yes, @leighman. I will be rolling back this change soon and include the appropriate documentation for typescript

**Comment by leighman** [comment]
I fixed this with https://github.com/react-community/lottie-react-native/pull/402. More documentation would probably be good.
